src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs

```diff
@@ -517,7 +517,7 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<T> ExcludeFromManifest<T>(this IResourceBuilder<T> builder) where T : IResource
     {
-        foreach (var annotation in builder.Resource.Annotations.OfType<ManifestPublishingCallbackAnnotation>())
+        foreach (var annotation in builder.Resource.Annotations.OfType<ManifestPublishingCallbackAnnotation>().ToArray())
```

Comment:
I think this whole block can now be replaced with:

```csharp
WithManifestPublishingCallback(ManifestPublishingCallbackAnnotation.Ignore)
```

Because we have the logic now for substituting rather than appending the manifest publishing callback.